docs: Restructure quick-start section of docs to separate clp-json from clp-text (#968)
y-scope/clp: docs/src/user-guide/quick-start/clp-json.md

### Queries

Regardless of what method you use to search, you'll need a query to find the logs you're looking for. Queries for JSON logs take the general form of

```bash
key: value
```

where `key` is the name of the JSON key you'd like to search within, and `value` is a sequence of characters you're looking for within that key. Multiple key-value pairs can be chained together with `AND`, `OR`, or `NOT`, like so:

```bash
key1: value1 AND key2: value2 OR key3: value3 ...
```

There are a number of other JSON-specific syntax rules that you can use to make your searches more powerful and effective. You can read about these syntax rules [here](../reference-json-search-syntax).

### Searching from the command line

If you'd like to search your query from the command line, run the following command from inside the package:

```bash
sbin/search.sh '<query>'
```

To narrow your search to a specific time range:

* Add `--begin-time <epoch-timestamp-millis>` to filter for log events after a certain time.
    * `<epoch-timestamp-millis>` is the timestamp as milliseconds since the UNIX epoch.
* Add `--end-time <epoch-timestamp-millis>` to filter for log events after a certain time.

> 

Author: oh @kirkrodrigues I didn't realize that the "narrow your search to a specific time range" section actually says "after a certain time" for both `--begin-time <epoch-timestamp-millis>` and `--end-time <epoch-timestamp-millis>`. Does the `--begin-time` command filter for after a certain time and the `--end-time` filter for before a certain time? or is it the other way around?

# clp-json quick-start

This page will walk you through how to start CLP and use it to compress and search JSON logs.

:::{caution}

> If you're using a `clp-json` release, you can only compress and search JSON logs.

Reviewer: ```suggestion
If you're using a `clp-json` release, you can only compress and search JSON logs. This limitation
will be addressed in a future version of CLP.
```

:::

---

## Starting CLP

To start CLP, run:

```bash
sbin/start-clp.sh
```

:::{note}
If CLP fails to start (e.g., due to a port conflict), try adjusting the settings in
`etc/clp-config.yml` and then run the start command again.
:::

---

## Compressing JSON logs

To compress some JSON logs, run:

```bash
sbin/compress.sh --timestamp-key '<timestamp-key>' <path1> [<path2> ...]
```

* `<timestamp-key>` is the field path of the kv-pair that contains the timestamp in each log event.
  * E.g., if your log events look like
    `{"timestamp": {"iso8601": "2024-01-01 00:01:02.345", ...}}`, you should enter
    `timestamp.iso8601` as the timestamp key.

  :::{caution}
  Log events without the specified timestamp key will *not* have an assigned timestamp. Currently,
  these events can only be searched from the command line (when you don't specify a timestamp
  filter).
  :::

* `<path...>` are paths to JSON log files or directories containing such files.

>   * Each JSON log file should contain each log event as a separate
>     [JSON object](./index.md#clp-json), i.e., *not* as an array.

Reviewer: ```suggestion
  * Each JSON log file should contain each log event as a 
    [separate JSON object](./index.md#clp-json), i.e., *not* as an array.
```

> 

Reviewer: ```suggestion

The compression script will output the compression ratio of each dataset you compress, or you can
use the UI to view overall statistics.

```

Author: I changed "UI" to "web UI" --- @kirkrodrigues lemme know if you aren't ok with that

Author: nvm I'm just gonna do UI like you said originally

Compressed logs will be stored in the directory specified by the `archive_output.storage.directory`
config option in `etc/clp-config.yml` (`archive_output.storage.directory` defaults to
`var/data/archives`).

:::{tip}
To compress logs from object storage, see
[Using object storage](../guides-using-object-storage/index).
:::

### Sample logs

For some sample logs, check out the open-source [datasets](../resources-datasets).

> ### Examining compression statistics
> 
> The compression script used above will output the compression ratio of each dataset you compress,
> or you can use the UI to view overall statistics.
> 

Reviewer: ```suggestion
```

Replaced by the above.

---

## Searching JSON logs

> You can search through your logs using queries from the UI or from the command line.
> 
> ### Queries
> 
> Regardless of what method you use to search, you'll need a query to find the logs you're looking
> for. Queries for JSON logs take the general form of
> 
> ```bash
> key: value
> ```
> 
> where `key` is the name of the JSON key you'd like to search within, and `value` is a sequence of
> characters you're looking for within that key. Multiple key-value pairs can be chained together
> with `AND`, `OR`, or `NOT`, like so:
> 
> ```bash
> key1: value1 AND key2: value2 OR key3: value3 ...
> ```
> 
> There are a number of other JSON-specific syntax rules that you can use to make your searches more
> powerful and effective. You can read about these rules on the
> [JSON syntax reference page](../reference-json-search-syntax).

Reviewer: ````suggestion
You can search your compressed logs from CLP's [UI](#searching-from-the-ui) or the
[command line](#searching-from-the-command-line).

Queries must be in the form of a set of conditions (predicates) on key-value pairs (kv-pairs). For
example, consider the logs in [Figure 1](#figure-1) and the query in [Figure 2](#figure-2).

(figure-1)=
:::{card}

```json lines
{
  "t": {
    "$date": "2023-03-21T23:46:37.392"
  },
  "ctx": "conn11",
  "msg": "Waiting for write concern."
}
{
  "t": {
    "$date": "2023-03-21T23:46:37.392"
  },
  "msg": "Set last op to system time"
}
```

+++
**Figure 1**: A set of JSON log events.
:::

(figure-2)=
:::{card}

```sql
ctx: "conn11" AND msg: "*write concern*"
```

+++
**Figure 2**: An example query.
:::

The query in [Figure 2](#figure-2) will match log events that contain the kv-pair `"ctx": "conn11"`
as well as a kv-pair with key `"msg"` and a value that matches the wildcard query
`"*write concern*"`. This query will match the first log event in [Figure 1](#figure-1).

A complete reference for clp-json's query syntax is available on the
[JSON syntax reference page](../reference-json-search-syntax).
````

> ### Searching from the command line

Reviewer: Can we swap the UI section and this section?

> If you'd like to search your query from the command line, run the following command from inside the
> package:

Reviewer: ```suggestion
To search your compressed logs from the command line, run:
```

```bash
sbin/search.sh '<query>'
```

To narrow your search to a specific time range:

* Add `--begin-time <epoch-timestamp-millis>` to filter for log events after a certain time.
  * `<epoch-timestamp-millis>` is the timestamp as milliseconds since the UNIX epoch.
* Add `--end-time <epoch-timestamp-millis>` to filter for log events before a certain time.

To perform case-insensitive searches, add the `--ignore-case` flag.

:::{caution}
To match the convention of other tools, by default, searches are case-**insensitive** in the UI and
searches are case-**sensitive** on the command line.
:::

### Searching from the UI

> If you'd like to search your query from the web UI, CLP includes a web interface available at
> [http://localhost:4000](http://localhost:4000) by default (if you changed `webui.host` or
> `webui.port` in `etc/clp-config.yml`, use the new values).

Reviewer: ```suggestion
To search your compressed logs from CLP's UI, open [http://localhost:4000](http://localhost:4000) in
your browser (if you changed `webui.host` or `webui.port` in `etc/clp-config.yml`, use the new
values).
```

:::{image} clp-search-ui.png
:::

The image above shows the search page after running a query. The numbered circles correspond to
the following features:

1. The search box is where you can enter your query.
2. The timeline shows the number of results across the time range of your query.
   * You can click and drag to zoom into a time range, or use the time range filter in (4).
3. The table displays the search results for your query.
4. Clicking the <i class="fa fa-bars"></i> icon reveals additional filters for your query.
   * The time range filter allows you to specify the period of time that matching log events must be
     in.
   * The case sensitivity filter allows you to specify whether CLP should respect the case of your
     query.
5. Clicking the <i class="fa fa-cog"></i> icon reveals options for displaying results.
6. The <i class="fa fa-trash"></i> icon clears the results of the last query.

:::{note}
By default, the UI will only return 1,000 of the latest search results. To perform searches which
return more results, use the [command line](#searching-from-the-command-line).
:::

---

## Stopping CLP

If you need to stop CLP, run:

```bash
sbin/stop-clp.sh
```

> 
> ---
> 
> ## More information
> 
> You've reached the end of the clp-json quick-start guide. For more information on clp-json,
> visit the [CLP for JSON logs](../core-clp-s) page.

Reviewer: ```suggestion
```

I think this might confuse users who get this far since clp-s and clp-json are different beasts.

Author: oh --- I thought `clp-json` and `clp-s` were the same thing with different names? cause you use the `clp-s` binaries to compress and search JSON logs

Reviewer: `clp-json` refers to the package which contains `clp-s` as well as everything else in the package (e.g., the query scheduler). `clp-s` refers to the single core binary.

Author: gotcha thank you!

```

* `<timestamp-key>` is the field path of the kv-pair that contains the timestamp in each log event.
  * E.g., if your log events look like
    `{"timestamp": {"iso8601": "2024-01-01 00:01:02.345", ...}}`, you should enter
    `timestamp.iso8601` as the timestamp key.

  :::{caution}
  Log events without the specified timestamp key will *not* have an assigned timestamp. Currently,
  these events can only be searched from the command line (when you don't specify a timestamp
  filter).
  :::

* `<path...>` are paths to JSON log files or directories containing such files.
  * Each JSON log file should contain each log event as a
    [separate JSON object](./index.md#clp-json), i.e., *not* as an array.

The compression script will output the compression ratio of each dataset you compress, or you can
use the UI to view overall statistics.

Compressed logs will be stored in the directory specified by the `archive_output.storage.directory`
config option in `etc/clp-config.yml` (`archive_output.storage.directory` defaults to
`var/data/archives`).

:::{tip}
To compress logs from object storage, see
[Using object storage](../guides-using-object-storage/index).
:::

### Sample logs

> For some sample logs, check out the open-source [datasets](../resources-datasets).

Reviewer: ```suggestion
For some sample logs, check out the [open-source datasets](../resources-datasets).
```

---

## Searching JSON logs

You can search your compressed logs from CLP's [UI](#searching-from-the-ui) or the
[command line](#searching-from-the-command-line).

> Queries must be in the form of a set of conditions (predicates) on key-value pairs (kv-pairs). For
> example, consider the logs in [Figure 1](#figure-1) and the query in [Figure 2](#figure-2).

Reviewer: ```suggestion
In clp-json, queries are written as a set of conditions (predicates) on key-value pairs (kv-pairs).
For example, [Figure 1](#figure-1) shows a query that matches the first log event in
[Figure 2](#figure-2).
```

Edited based on your writing in clp-text.md. Can you swap the content of the figures?

(figure-1)=
:::{card}

```json lines
{
  "t": {
    "$date": "2023-03-21T23:46:37.392"
  },
  "ctx": "conn11",
  "msg": "Waiting for write concern."
}
{
  "t": {
    "$date": "2023-03-21T23:46:37.392"
  },
  "msg": "Set last op to system time"
}
```

+++
**Figure 1**: A set of JSON log events.
:::

(figure-2)=
:::{card}

```sql
ctx: "conn11" AND msg: "*write concern*"
```

+++
**Figure 2**: An example query.
:::

The query in [Figure 2](#figure-2) will match log events that contain the kv-pair `"ctx": "conn11"`
as well as a kv-pair with key `"msg"` and a value that matches the wildcard query

> `"*write concern*"`. This query will match the first log event in [Figure 1](#figure-1).

Reviewer: ```suggestion
`"*write concern*"`.
```

A complete reference for clp-json's query syntax is available on the
[JSON syntax reference page](../reference-json-search-syntax).

### Searching from the UI

To search your compressed logs from CLP's UI, open [http://localhost:4000](http://localhost:4000) in
your browser (if you changed `webui.host` or `webui.port` in `etc/clp-config.yml`, use the new
values).

:::{image} clp-search-ui.png
:::

The image above shows the search page after running a query. The numbered circles correspond to
the following features:

1. The search box is where you can enter your query.
2. The timeline shows the number of results across the time range of your query.
   * You can click and drag to zoom into a time range, or use the time range filter in (4).
3. The table displays the search results for your query.
4. Clicking the <i class="fa fa-bars"></i> icon reveals additional filters for your query.

>    * The time range filter allows you to specify the period of time that matching log events must
>      be in.

Reviewer: ```suggestion
   * The time range filter allows you to specify the period of time that matching log events must be
     in.
```

Reviewer: It's exactly 100 characters when I count it.

Author: ok!

   * The case sensitivity filter allows you to specify whether CLP should respect the case of your
     query.
5. Clicking the <i class="fa fa-cog"></i> icon reveals options for displaying results.
6. The <i class="fa fa-trash"></i> icon clears the results of the last query.

:::{note}
By default, the UI will only return 1,000 of the latest search results. To perform searches which
return more results, use the [command line](#searching-from-the-command-line).
:::

> ### Searching from the command line

Author: this is not an extra blank line.

# clp-json quick-start

This page will walk you through how to start CLP and use it to compress and search JSON logs.

:::{caution}
If you're using a `clp-json` release, you can only compress and search JSON logs. This limitation
will be addressed in a future version of CLP.
:::

---

## Starting CLP

To start CLP, run:

```bash
sbin/start-clp.sh
```

:::{note}
If CLP fails to start (e.g., due to a port conflict), try adjusting the settings in
`etc/clp-config.yml` and then run the start command again.
:::

---

## Compressing JSON logs

To compress some JSON logs, run:

```bash
sbin/compress.sh --timestamp-key '<timestamp-key>' <path1> [<path2> ...]
```

* `<timestamp-key>` is the field path of the kv-pair that contains the timestamp in each log event.
  * E.g., if your log events look like
    `{"timestamp": {"iso8601": "2024-01-01 00:01:02.345", ...}}`, you should enter
    `timestamp.iso8601` as the timestamp key.

  :::{caution}
  Log events without the specified timestamp key will *not* have an assigned timestamp. Currently,
  these events can only be searched from the command line (when you don't specify a timestamp
  filter).

>   :::

Author: there are empty lines before and after all caution statements in clp-json.md

:::

### Sample logs

For some sample logs, check out the [open-source datasets](../resources-datasets).

---

## Searching JSON logs

You can search your compressed logs from CLP's [UI](#searching-from-the-ui) or the
[command line](#searching-from-the-command-line).

In clp-json, queries are written as a set of conditions (predicates) on key-value pairs (kv-pairs).
For example, [Figure 1](#figure-1) shows a query that matches the first log event in
[Figure 2](#figure-2).

(figure-1)=
:::{card}

```sql
ctx: "conn11" AND msg: "*write concern*"
```

+++
**Figure 1**: An example query.
:::

(figure-2)=
:::{card}

```json lines
{
  "t": {
    "$date": "2023-03-21T23:46:37.392"
  },
  "ctx": "conn11",
  "msg": "Waiting for write concern."
}
{
  "t": {
    "$date": "2023-03-21T23:46:37.392"
  },
  "msg": "Set last op to system time"
}
```

+++
**Figure 2**: A set of JSON log events.
:::

The query in [Figure 2](#figure-2) will match log events that contain the kv-pair `"ctx": "conn11"`
as well as a kv-pair with key `"msg"` and a value that matches the wildcard query
`"*write concern*"`.

A complete reference for clp-json's query syntax is available on the

> [JSON syntax reference page](../reference-json-search-syntax).

Reviewer: ```suggestion
[syntax reference page](../reference-json-search-syntax).
```

Technically, JSON has its own syntax, so "JSON syntax" might sound misleading.
